server/chat-commands/core.ts

```diff
@@ -349,29 +348,30 @@ exports.commands = {
 	authlist: 'authority',
 	authority(target, room, user, connection) {
 		if (target) {
-			let targetRoom = Rooms.search(target);
-			let availableRoom = targetRoom && targetRoom.checkModjoin(user);
+			const targetRoom = Rooms.search(target);
+			const availableRoom = targetRoom?.checkModjoin(user);
 			if (targetRoom && availableRoom) return this.parse(`/roomauth1 ${target}`);
 			return this.parse(`/userauth ${target}`);
 		}
 		/** @type {{[k: string]: string[]}} */
-		let rankLists = {};
-		let ranks = Object.keys(Config.groups);
-		for (let u in Users.usergroups) {
-			let rank = Users.usergroups[u].charAt(0);
+		const rankLists: AnyObject = {};
+		const ranks = Object.keys(Config.groups);
+		for (const u in Users.usergroups) {
+			const rank = Users.usergroups[u].charAt(0);
 			if (rank === ' ' || rank === '+') continue;
 			// In case the usergroups.csv file is not proper, we check for the server ranks.
 			if (ranks.includes(rank)) {
-				let name = Users.usergroups[u].substr(1);
-				if (!rankLists[rank]) rankLists[rank] = [];
+				const name: string = Users.usergroups[u].substr(1);
+				const place: string[] = rankLists[rank];
+				if (!place) rankLists[rank] = [];
 				if (name) rankLists[rank].push(name);
 			}
 		}
 
-		let buffer = Object.keys(rankLists).sort(
+		const buffer = Object.keys(rankLists).sort(
 			(a, b) => (Config.groups[b] || {rank: 0}).rank - (Config.groups[a] || {rank: 0}).rank
 		).map(
-			r => `${(Config.groups[r] ? `**${Config.groups[r].name}s** (${r})` : r)}:\n${rankLists[r].sort((a, b) => toID(a).localeCompare(toID(b))).join(", ")}`
+			r => `${(Config.groups[r] ? `**${Config.groups[r].name}s** (${r})` : r)}:\n${rankLists[r].sort((a: string, b: string) => toID(a).localeCompare(toID(b))).join(", ")}`
```

Comment:
```suggestion
			r => `${(Config.groups[r] ? `**${Config.groups[r].name}s** (${r})` : r)}:\n${rankLists[r].sort((a, b) => toID(a).localeCompare(toID(b))).join(", ")}`
```

TS has type inference for this. If you need to add types to an arrow function, you're doing something wrong (in this case, the wrong thing you're doing is failing to correctly type `rankLists`.